src/poetry/publishing/uploader.py

```diff
@@ -275,6 +285,12 @@ def _upload_file(
                 elif resp.status_code == 400 and "was ever registered" in resp.text:
                     self._register(session, url)
                     resp.raise_for_status()
+                elif skip_existing and self._is_file_exists_error(resp):
+                    self._io.overwrite(
+                        f" - Uploading <c1>{file.name}</c1> <warning>File exists."
+                        " Skipping</>"
+                    )
+                    bar.finish()
```

Comment:
```suggestion
                    bar.set_format(
                        f" - Uploading <c1>{file.name}</c1> <warning>File exists."
                        " Skipping</>"
                    )
                    bar.display()
```
If `overwrite()` is called on an undecorated IO, the output will contain control characters which can't be displayed properly. Further, I think the message should go to error instead of out because all progress bar output goes to error and I'd consider it part of the progress bar output. Maybe we can revisit your previous solution and tweak it a bit:
If we call `diplay()` instead of `finish()`, it should work because we circumvent the early return.

Reply:
ah, thanks